File: python/1545.py
def rev(b):
    ret = ""
    for i in b:
        if i == '0':
            ret += '1'
        else:
            ret += '0'
    return ret[::-1]

class Solution:
    def findKthBit(self, n: int, k: int) -> str:
        s = []
        for i in range(n):
            if i == 0:
                s.append("0")
                continue
            s.append(s[i-1] + "1" + rev(s[i-1]))
        return s[len(s)-1][k-1]

File: rust/1545.rs
fn rev(n: String) -> String {
    let mut b = String::new();
    for i in n.chars().rev() {
        if i == '0' {
            b += "1";
        } else {
            b += "0";
        }
    }
    b
}
impl Solution {
    pub fn find_kth_bit(n: i32, k: i32) -> char {
        let mut s = Vec::new();
        for i in 0..n {
            if i == 0 {
                s.push("0".to_string());
            } else {
                let sc = s.clone();
                let f = format!("{}1{}",sc[(i-1) as usize],rev(sc[(i-1) as usize].to_string())).to_string();
                s.push(f.clone());
            }
        }
        let e = &s[s.clone().len()-1 as usize].chars().collect::<Vec<_>>();
        e[(k-1) as usize]
    }
}
